### DISCUSSION TITLE
Test cases in OATS

### DISCUSSION THREAD

**Discussion by bukhsh** [bukhsh/oats#103] [Ideas]
@calume do you have any network data that you'd like to share?

I've added a section in OATS documentation, see [here](https://oats.readthedocs.io/en/latest/networkdata.html)

**Comment by calume** [comment]
Hi Waqquas,
I can add the Cornwall network if you want, ive used many enough times as you know and is an MV/HV/EHV network example, could even include the version with stitched on 11 kV Whitchurch network.

On another subject... do you want to help create an official release for the github of the EV optimisation model? We could work from James' double sided price version, but we would need to fix a few bugs that we fixed before for my version (for the headroom paper). I am working off my custom version which only works on python 3.7 and a specific version of Pyomo, I have released this to an AGILE partner who is happy working with it (once he got it working!, hopefully there will be a paper in this eventually and we will reference OATS) and now I am trying to get it working with Stuarts current 4th year student (I gave him a requirements.txt file of all packagas and versions that I had tested and works on my PC, but guess what-he has an apple mac-ARGGHHH). Clearly it needs to work on the latest versions of python and pyomo which might not be too difficult but that's where I could do with your help. The same applies for my DSO-TSO coordination models which I should really be making open source, although I have a bad feeling the PWL formulation does not work on the latest version of pyomo.

Also I know you have a storage model, but what about a heat storage / heat pump model? This would be a longer term project, perhaps not what you want OATS to do, especially considering this would involve LV networks (although you could just stick to single bus model for these). Just an idea, im sure you have enough on your plate. If a hydrogen electolyser model could also be integrated we could be looking at a whole system energy model which I assume will become the standard in years to come.

Anyways, I will have a look into an .xlsx file for the Cornwall network, which I will convert to the latest format from the github and send to you in the coming weeks,

Cheers

Calum

**Reply by bukhsh** [reply]
Thanks, Calum.
I had forgotten about the Cornwall network. If you could add it to the test cases folder, that'd be great. Let me know if you need any help with the data format. It'd be good to release the 11kV as a separate test case with a note that it could be bolt onto the Cornwall network. We can only release OATS test vases that are built using publically available data - I think that is the case with the Cornwall network.

I have fixed several issues in OATS to make it compatible with the recent release of Pyomo. The version on Github is the latest - I need to remind myself how to push these changes to the 'pip' downloadable thing.

Well done to your AGILE partner for getting the EV optimisation working. I think it's great that researchers outside of Strathclyde are finding value in our models. 

Regarding a GitHub release for EV optimisation, I think it'd be good to release it with a paper (maybe in a journal that is less painful to publish than IEEE Trans on Power Systems) - what do you think?

**Comment by calume** [comment]
Yes a paper for the release of OATS would be fine, it was ofcourse used in James' paper, and all being my headroom paper too.

I was impressed by Davids (student I am working with) plan for car park EV optimisation. Slight potential for a paper there perhaps...

What could we add to the EV optimisation that would make it worthy of its own paper?

Cheers
Calum
